nalgebra-sparse/src/pattern.rs

```diff
@@ -184,6 +186,131 @@ impl SparsityPattern {
         })
     }
 
+    /// Try to construct a sparsity pattern from the given dimensions, major offsets
+    /// and minor indices.
+    ///
+    /// Returns an error if the data does not conform to the requirements.
+    pub fn from_offset_and_indices_unchecked(
+        major_dim: usize,
+        minor_dim: usize,
+        major_offsets: Vec<usize>,
+        minor_indices: Vec<usize>,
+    ) -> Result<Self, SparsityPatternFormatError> {
```

Comment:
This should be an `unsafe` method, because if the variants are not upheld downstream code may lead to undefined behavior. Moreover, I don't think we should return a `Result` here: simply panic (by way of assertions) if the most basic invariants are not upheld.